app/controllers/checkout/discounts_controller.rb

```diff
@@ -104,8 +107,8 @@ def clean_params
     end
 
     def parse_date_times
-      offer_code_params[:valid_at] = Date.parse(offer_code_params[:valid_at]) if offer_code_params[:valid_at].present?
-      offer_code_params[:expires_at] = Date.parse(offer_code_params[:expires_at]) if offer_code_params[:expires_at].present?
+      offer_code_params[:valid_at] = Time.zone.parse(offer_code_params[:valid_at]) if offer_code_params[:valid_at].present?
+      offer_code_params[:expires_at] = Time.zone.parse(offer_code_params[:expires_at]) if offer_code_params[:expires_at].present?
```

Comment:
Use `Time.zone.parse` instead of `Date.parse` to preserve time information. Previously, datetime inputs like "2022-11-02T23:00" were being converted to midnight UTC, losing the time component entirely. This fixes a production bug where discount validity times were being ignored.